alibi_detect/saving/_tensorflow/tests/test_saving_tf.py

```diff
@@ -1,26 +1,78 @@
 from pytest_cases import param_fixture, parametrize, parametrize_with_cases
+import pytest
 
 from alibi_detect.saving.tests.datasets import ContinuousData
 from alibi_detect.saving.tests.models import encoder_model
 
 from alibi_detect.cd.tensorflow import HiddenOutput as HiddenOutput_tf
 from alibi_detect.saving.loading import _load_model_config, _load_optimizer_config
-from alibi_detect.saving.saving import _path2str, _save_model_config
-from alibi_detect.saving.schemas import ModelConfig
+from alibi_detect.saving.saving import _path2str, _save_model_config, _save_optimizer_config
+from alibi_detect.saving.schemas import ModelConfig, SupportedOptimizer
 import tensorflow as tf
+import numpy as np
+from packaging import version
 
 backend = param_fixture("backend", ['tensorflow'])
 
 
 # Note: The full save/load functionality of optimizers (inc. validation) is tested in test_save_classifierdrift.
-def test_load_optimizer_object(backend):
+@pytest.mark.skipif(version.parse(tf.__version__) < version.parse('2.11.0'),
+                    reason="Skipping since tensorflow < 2.11.0")
```

Comment:
@jklaise this is admittedly a bit weird. The new testing logic assumed tf 2.11 was being used, so tests would have failed on old tensorflow versions. I've added two separate tests for tf>=2.11 and tf<2.11, so that we can test for backward compatibility with old tensorflow versions when desired, but this would have to be by manually installing tf<2.11 locally, since CI will install the latest version. 

Since we don't test old versions of deps anywhere else, I'm wondering if this is overkill, or if there's no harm in keeping both tests...

Reply:
I see, I think it's fine to leave this in so we can debug locally with older `tensorflow` versions if required.